test/suites/para/test_tanssi_containers.ts

```diff
@@ -159,5 +167,165 @@ describeSuite({
+              key: "Vec<u8>",
+              value: "Vec<u8>",
+            }
+          },
+          rpc: {
+            utils: {
+              raw_chain_spec_into_container_chain_genesis_data: {
+                description: 'Convert a raw chain spec string into a ContainerChainGenesisData',
+                params: [
+                  {
+                    name: 'raw_chain_spec',
+                    type: 'Text'
+                  }
+                ],
+                type: '(u32, ContainerChainGenesisData)'
+              }
+            }
+          }
+        });
+        let spec2002text = paraApi2.createType('Text', spec2002);
+        const containerChainGenesisDataFromRpc = await paraApi2.rpc.utils.raw_chain_spec_into_container_chain_genesis_data(spec2002text);
+        expect(containerChainGenesisDataFromRpc[0].toNumber()).to.be.equal(2002);
+
+        // Before registering container chain 2002, ensure that it has 0 blocks
+        // Since the RPC doesn't exist at this point, we need to get that from the relay
+        const header2002 = await getHeaderFromRelay(relayApi, 2002);
+        expect(header2002.number.toNumber()).to.be.equal(0);
+        const registered1 = (await paraApi.query.registrar.registeredParaIds());
+        expect(registered1.toJSON().includes(2002)).to.be.false;
+
+        const tx = paraApi.tx.registrar.register(2002, containerChainGenesisDataFromRpc[1]);
+        await paraApi.tx.sudo.sudo(tx).signAndSend(alice);
+        
+        const tanssiBlockNum = (await paraApi.rpc.chain.getBlock()).block.header.number.toNumber();
+        // TODO: this should wait 2 sessions. We are waiting 10 + 20 blocks
+        await countUniqueBlockAuthors(context, paraApi, 10, 4);
+        await context.waitBlock(20, "Tanssi");
+
+        // Check that pending para ids contains 2002
+        const registered = (await paraApi.query.registrar.registeredParaIds());
+        expect(registered.toJSON().includes(2002)).to.be.true;
+
+        // This ws api is only available after the node detects its assignment
+        // TODO: wait up to 30 seconds after a new block is created to ensure this port is available
+        const wsProvider = new WsProvider('ws://127.0.0.1:9951');
+        let container2002Api = await ApiPromise.create({ provider: wsProvider });
+
+        const container2002Network = container2002Api.consts.system.version.specName.toString();
+        const paraId2002 = (await container2002Api.query.parachainInfo.parachainId()).toString();
+        expect(container2002Network, "Container2002 API incorrect").to.contain("container-chain-template");
+        expect(paraId2002, "Container2002 API incorrect").to.be.equal("2002");
+
+        // Check authors of tanssi blocks
+        // Should be 2 different keys because 2002 has been registered
+        await countUniqueBlockAuthors(context, paraApi, 4, 2);
+
+        let blockNum = (await container2002Api.rpc.chain.getBlock()).block.header.number.toNumber();
```

Comment:
Isnt this essentially the same as waitForBlock?

Reply:
Ideally yes, but:

    // Cannot use context.waitBlock because the container2002Api is not part of moonwall

I need to see if it is possible to add a network to the moonwall context while running a test, or some other workaround.